File: backend/app/models/document.py
"""
Document Analysis Response Models
"""
from pydantic import BaseModel, Field
from typing import List, Dict, Optional, Any


class ClausePositionResponse(BaseModel):
    """Clause position data for frontend"""
    clause_text: str
    page_number: int
    start_char: int
    end_char: int
    risk_score: float
    risk_severity: str
    risk_category: str
    risk_explanation: str
    recommended_action: str
    confidence: float
    bounding_box: Optional[Dict[str, float]] = None
    validation_result: Optional[Dict[str, Any]] = None  # Validation result from validator agent


class PDFMetadata(BaseModel):
    """PDF file metadata"""
    total_pages: int
    file_size: int
    filename: str


class ClauseHighlightResponse(BaseModel):
    """Complete response for document analysis with clause highlighting"""
    risk_assessment: Dict[str, Any]  # Full risk assessment data
    clause_positions: List[ClausePositionResponse]
    pdf_metadata: PDFMetadata
    pdf_base64: str = Field(..., description="Base64-encoded PDF for frontend rendering")

File: frontend/types/document.ts
/*
 * TypeScript interfaces for document analysis with clause highlighting
 */
import { ValidationResult } from './validation';

export interface ClausePosition {
    clause_text: string;
    page_number: number;
    start_char: number;
    end_char: number;
    risk_score: number;
    risk_severity: string;
    risk_category: string;
    risk_explanation: string;
    recommended_action: string;
    confidence: number;
    validation_result?: ValidationResult;  // NEW: Optional validation result
    bounding_box?: {
        x: number;
        y: number;
        width: number;
        height: number;
    };
}

export interface PDFMetadata {
    total_pages: number;
    file_size: number;
    filename: string;
}

export interface ClauseHighlightData {
    risk_assessment: any; // Risk assessment data (matches backend structure)
    clause_positions: ClausePosition[];
    pdf_metadata: PDFMetadata;
    pdf_base64: string;
}

// Helper type: clauses grouped by page number
export type ClausesByPage = Record<number, ClausePosition[]>;
